src/snovault/elasticsearch/indexer.py

```diff
@@ -143,140 +154,302 @@ def get_related_uuids(request, es, updated, renamed):
+        remote_indexing_threshold = default_threshold
+    if uuid_count < default_threshold:
+        return False
+    return True
+
+
+def _get_sqs_arn(queue_name):
+    boto_session = boto3.Session(region_name = 'us-west-2')
+    sqs = boto_session.resource('sqs')
+    try:
+        queue = sqs.get_queue_by_name(QueueName=queue_name)
+        return queue.url
+    except Exception as ecp:
+        log.warning(f"Remote indexing _get_sqs_arn exception: {repr(ecp)}")
+        return None
+
+
+def _send_sqs_msg(
+        cluster_name,
+        remote_indexing,
+        body,
+        delay_seconds=0,
+        save_event=False,
+    ):
+    error_str = None
+    msg_attrs = copy.deepcopy(_SQS_INPUT_TEMPATE)
+    msg_attrs['cluster_name'] = cluster_name
+    msg_attrs['remote_indexing'] = '0' if remote_indexing else '1'
+    msg_attrs['save_event'] = '0' if save_event else '1'
+    msg_attrs = {
+        key: {'DataType': 'String', 'StringValue': val}
+        for key, val in msg_attrs.items()
+    }
+    queue_arn = _get_sqs_arn('ec2_scheduler_input')
+    response = _SQS.send_message(
+        QueueUrl=queue_arn,
+        DelaySeconds=delay_seconds,
+        MessageAttributes=msg_attrs,
+        MessageBody=body,
+    )
+    log.warning(f"Remote indexing _send_sqs_msg: {body}")
+    status_code = response['ResponseMetadata']['HTTPStatusCode']
+    return status_code
+
+
+def _get_nodes(request, indexer_state):
+    label = 'nothing'
+    continue_on = False
+    did_timeout = False
+    # Get local/remote indexing state
+    head_node, indexing_node, time_now, did_fail, is_indexing_node = setup_indexing_nodes(
+        request, indexer_state
+    )
+    this_node = head_node
+    other_node = indexing_node
+    if is_indexing_node:
+        this_node = indexing_node
+        other_node = head_node
+    if did_fail:
+        did_timeout = True
+        return this_node, other_node, continue_on, did_timeout
+    this_time_in_state = -1
+    other_time_in_state = -1
+    this_time_in_state = time_now - float(this_node['last_run_time'])
+    other_time_in_state = time_now - float(other_node['last_run_time'])
+    # Check for early return if indexing is running
+    if this_node['node_index'] == HEAD_NODE_INDEX:
+        if other_node['done_indexing']:
+            label = 'Head node with remote finished indexing.  Reset both es node states.'
+            this_node, other_node, time_now, did_fail, is_indexing_node = setup_indexing_nodes(
+                request,
+                indexer_state,
+                reset=True
+            )
+        elif other_node['started_indexing']:
+            label = 'Head node with remote indexing running'
+        elif this_node['waiting_on_remote']:
+            label = 'Head node waiting on remote indexing to start'
+            if this_time_in_state > 60*10:
+                # indexer machine did not start, move on without it
+                # need to cancel it?
+                did_timeout = True
```

Comment:
Need to handle canceling the indexer instance.  The indexer could be coming up or the message in flight or not yet sent to the lambda function, etc.  

- Might be able to store the receipt handle in the node state and delete if possible. 
- We have the indexer aws instance state so check that too and maybe reset the clock if pending.
- Or just should reseting the index node state might handle it all.

This should be blocking too.